pybamm/models/submodels/particle_cracking/base_cracking.py

```diff
@@ -0,0 +1,156 @@
+    Parameters
+    ----------
+    param : parameter class
+        The parameters to use for this submodel
+    domain : dict, optional
+        Dictionary of either the electrode for "Positive" or "Nagative"
+    **Extends:** :class:`pybamm.BaseSubModel`
+
+    Reference
+    Ai, W., Kraft, L., Sturm, J., Jossen, A., & Wu, B. (2019). Electrochemical
+    Thermal-Mechanical Modelling of Stress Inhomogeneity in Lithium-Ion Pouch
+    Cells. Journal of The Electrochemical Society, 167(1), 013512.
+    """
+
+    def __init__(self, param, domain):
+        self.domain = domain
+        super().__init__(param)
+
+    def _get_standard_variables(self, l_cr):
+        domain = self.domain.lower() + " particle"
+        if self.domain == "Positive":
+            l_cr0 = self.param.l_cr_p_0
+        else:
+            l_cr0 = self.param.l_cr_n_0
+        l_cr_av = pybamm.x_average(l_cr)
+        variables = {
+            self.domain + " particle crack length [m]": l_cr * l_cr0,
+            self.domain + " particle crack length": l_cr,
+            f"X-averaged {domain} crack length": l_cr_av,
+            f"X-averaged {domain} crack length [m]": l_cr_av * l_cr0,
+        }
+        variables.update(self._get_standard_surface_variables(l_cr))
+        return variables
+
+    def _get_mechanical_results(self, variables):
+        c_s_rav = variables[
+            "R-averaged " + self.domain.lower() + " particle concentration"
+        ]
+        c_s_surf = variables[self.domain + " particle surface concentration"]
+        T_xav = variables["X-averaged cell temperature"]
+        c_s_rxav = variables[
+            "Average " + self.domain.lower() + " particle concentration"
+        ]
+
+        if "Cell thickness change [m]" not in variables:
+            cell_thickness_change = (
+                T_xav * self.param.Delta_T * self.param.alpha_T_cell_dim
+            )  # thermal expansion
+        else:
+            cell_thickness_change = variables["Cell thickness change [m]"]
+
+        if self.domain == "Negative":
+            Omega = self.param.Omega_n
+            R0 = self.param.R_n
+            c_scale = self.param.c_n_max
+            c_0 = self.param.c_n_0
+            E0 = self.param.E_n
+            nu = self.param.nu_n
+            eps_s = self.param.epsilon_s_n
+            L0 = self.param.L_n
+            c_init = self.param.c_n_init(1)
+            v_change = pybamm.x_average(
+                self.param.t_n_change(c_s_rav)
+            ) - self.param.t_n_change(c_init)
+
+        elif self.domain == "Positive":
+            Omega = self.param.Omega_p
+            R0 = self.param.R_p
+            c_scale = self.param.c_p_max
+            c_0 = self.param.c_p_0
+            E0 = self.param.E_p
+            nu = self.param.nu_p
+            eps_s = self.param.epsilon_s_p
+            L0 = self.param.L_p
+            c_init = self.param.c_p_init(0)
+            v_change = self.param.t_p_change(c_s_rxav) - self.param.t_p_change(c_init)
```

Comment:
should this be `pybamm.x_average(self.param.t_n_change(c_s_rav) -self.param.t_n_change(c_init))`. since `t_change` is nonlinear the order of the average matters, and at the moment it is different in the positive and negative case

Reply:
I wrote like this, because it is linear for the thickness change in the positive electrode. But I agree with you that it is better to be consistent, otherwise it will be confusing.